toolchain/codegen/codegen.cpp

```diff
@@ -0,0 +1,65 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/codegen/codegen.h"
+
+#include <cstdio>
+
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/MC/TargetRegistry.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Target/TargetMachine.h"
+#include "llvm/Target/TargetOptions.h"
+#include "llvm/TargetParser/Host.h"
+
+namespace Carbon {
+
+void PrintAssemblyFromModule(llvm::Module& module) {
+  llvm::raw_ostream& error_stream = llvm::errs();
```

Comment:
Rather than using llvm::errs and llvm::outs directly, these should be parameters passed from the Driver, which has `output_stream_` and `error_stream_` as members that you can use for this purpose (i.e., take a `llvm::raw_ostream&` parameter for each, and pass them along). That way, the stream they're going to can easily be changed (which we use in certain tests, and may eventually want to do here).

Reply:
Also llvm::outs